### ISSUE TITLE
Update enum for Python 3.6

### ISSUE THREAD

**Issue by brettcannon** [python/typeshed#854]
Bunch of changes to the enum module in Python 3.6. The type hints are also missing things that pre-date 3.6 like `__members__`. (The ability to iterate is covered by python/mypy/issues/2305.)

**Comment by JelleZijlstra** [comment]
A bunch of new things were already added in bbc7594ec72854033049a3452ae5cc0f0e347ce6 and Guido seems to be working on the metaclass in bebb49a1f2c0caa963f1f69e1bea6f3f01b62b54. I added some missing methods in 43c48cc686f70fa4b56c01cc069110a4ed1b39c8 and I can add any missing functionality on the metaclass once Guido's commit lands.

**Comment by FichteFoll** [comment]
I still get this with mypy 0.521 and Python 3.6.2:

```py
import enum


class NetworkEventName(str, enum.Enum):
    CONNECTED = "connected"
    DISCONNECTED = "disconnected"
    CLOSE_REQUEST = "close_request"
    MESSAGE = "message"
    RAW_LINE = "raw_line"


print(NetworkEventName.__members__)

```

```
> mypy /tmp/enum_test.py
/tmp/enum_test.py:12: error: Type[NetworkEventName] has no attribute "__members__"
```

**Comment by gvanrossum** [comment]
Can someone please file a separate issue for that?

**Comment by gvanrossum** [comment]
Also may be it's easy to fix with a PR to typeshed?

**Comment by FichteFoll** [comment]
Created as #1590.
